src/parkapi_sources/converters/apcoa/converter.py

```diff
@@ -0,0 +1,70 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+import requests
+from validataclass.exceptions import ValidationError
+from validataclass.validators import DataclassValidator
+
+from parkapi_sources.converters.base_converter.pull import PullConverter
+from parkapi_sources.exceptions import ImportParkingSiteException
+from parkapi_sources.models import RealtimeParkingSiteInput, SourceInfo, StaticParkingSiteInput
+
+from .mapper import ApcoaMapper
+from .validators import ApcoaParkingSiteInput
+
+
+class ApcoaPullConverter(PullConverter):
+    _base_url = 'https://api.apcoa-services.com/carpark-dev/v4'
+    required_config_keys = ['PARK_API_APCOA_API_SUBSCRIPTION_KEY']
+
+    mapper = ApcoaMapper()
+    apcoa_parking_site_validator = DataclassValidator(ApcoaParkingSiteInput)
+
+    source_info = SourceInfo(
+        uid='apcoa',
+        name='APCOA-SERVICES API',
+        public_url='https://devzone.apcoa-services.com/',
+        has_realtime_data=False,  # ATM only static data can be called from the API
+    )
+
+    def get_static_parking_sites(self) -> tuple[list[StaticParkingSiteInput], list[ImportParkingSiteException]]:
+        static_parking_site_inputs: list = []
+        static_parking_site_errors: list[ImportParkingSiteException] = []
+
+        parking_site_dicts = self.get_data()
+
+        for parking_site_dict in parking_site_dicts.get('Results', []):
+            try:
+                parking_site_input: ApcoaParkingSiteInput = self.apcoa_parking_site_validator.validate(parking_site_dict)
+            except ValidationError as e:
+                static_parking_site_errors.append(
+                    ImportParkingSiteException(
+                        source_uid=self.source_info.uid,
+                        parking_site_uid=parking_site_dict.get('CarParkId'),
+                        message=f'validation error for data {parking_site_dict}: {e.to_dict()}',
+                    ),
+                )
+                continue
+
+            static_parking_site_inputs.append(
+                self.mapper.map_static_parking_site(parking_site_input),
+            )
+        return static_parking_site_inputs, static_parking_site_errors
+
+    def get_realtime_parking_sites(self) -> tuple[list[RealtimeParkingSiteInput], list[ImportParkingSiteException]]:
+        return [], []  # ATM only static data can be called from the API
+
+    def get_data(self) -> dict:
+        headers: dict[str, str] = {
+            'Cache-Control': 'no-cache',
+            'Ocp-Apim-Subscription-Key': self.config_helper.get('PARK_API_APCOA_API_SUBSCRIPTION_KEY'),
+        }
+
+        response = requests.get(
+            f'{self.config_helper.get("PARK_API_APCOA_URL", self._base_url)}/Carparks',
```

Comment:
Why load this from config, and why not using `source_info.source_url`?

Reply:
now changed!

Reply:
Why appending `/Carparks` here?

Reply:
I assumed the ```source_url``` is the main endpoint and for flexibility, i defined only the main endpoint in ```source_url```should the converter need to get realtime data from other endpoints of the API later in the future. I have however removed it for now.

Reply:
In theory correct thought, in practice: do you know any other information which is relevant for ParkAPI which is provided there?

Reply:
I just added tariff information, which i mapped to ```has_fee```. Aside this, I don't think there are other information for now.
**Short question**: there is a ```CarparkType``` with value of Multi Level Car Park (```MLCP```), which I mapped to ```OFF_STREET_PARKING_GROUND```. Is this correct?

Reply:
It's more `CAR_PARK`, because `OFF_STREET_PARKING_GROUND` is just a flat ground.